fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/IndexOptions.java

```diff
@@ -223,6 +224,143 @@ public class IndexOptions {
+     * satisfying the select-neighbor heuristic may get added back in to pad the set of neighbors if the new node would
+     * otherwise have too few neighbors (see {@link Config#getM()}). The default value is set to
+     * {@link Config#DEFAULT_KEEP_PRUNED_CONNECTIONS}. See {@link Config#isKeepPrunedConnections()}.
+     */
+    public static final String HNSW_KEEP_PRUNED_CONNECTIONS = "hnswKeepPrunedConnections";
+
+    /**
+     * HNSW-only: If sampling is necessary (currently iff {@link #HNSW_USE_RABITQ} is {@code "true"}), this option
+     * represents the probability of a vector being inserted to also be written into the samples subspace of the hnsw
+     * structure. The vectors in that subspace are continuously aggregated until a total {@link #HNSW_STATS_THRESHOLD}
+     * has been reached. The default value is set to {@link Config#DEFAULT_SAMPLE_VECTOR_STATS_PROBABILITY}. See
+     * {@link Config#getSampleVectorStatsProbability()}.
+     */
+    public static final String HNSW_SAMPLE_VECTOR_STATS_PROBABILITY = "hnswSampleVectorStatsProbability";
+
+    /**
+     * HNSW-only: If sampling is necessary (currently iff {@link #HNSW_USE_RABITQ} is {@code "true"}), this option
+     * represents the probability of the samples subspace to be further aggregated (rolled-up) when a new vector is
+     * inserted. The vectors in that subspace are continuously aggregated until a total
+     * {@link #HNSW_STATS_THRESHOLD} has been reached. The default value is set to
+     * {@link Config#DEFAULT_MAINTAIN_STATS_PROBABILITY}. See {@link Config#getMaintainStatsProbability()}.
+     */
+    public static final String HNSW_MAINTAIN_STATS_PROBABILITY = "hnswMaintainStatsProbability";
+
+    /**
+     * HNSW-only: If sampling is necessary (currently iff {@link #HNSW_USE_RABITQ} is {@code "true"}), this option
+     * represents the threshold (being a number of vectors) that when reached causes the stats maintenance logic to
+     * compute the actual statistics (currently the centroid of the vectors that have been inserted to far). The result
+     * is then inserted into the access info subspace of the index. The default value is set to
+     * {@link Config#DEFAULT_STATS_THRESHOLD}. See {@link Config#getStatsThreshold()}.
+     */
+    public static final String HNSW_STATS_THRESHOLD = "hnswStatsThreshold";
+
+    /**
+     * HNSW-only: Indicator if we should RaBitQ quantization. See {@link com.apple.foundationdb.rabitq.RaBitQuantizer}
+     * for more details. The default value is set to {@link Config#DEFAULT_USE_RABITQ}.
+     * See {@link Config#isUseRaBitQ()}.
+     */
+    public static final String HNSW_USE_RABITQ = "hnswUseRaBitQ";
+
+    /**
+     * HNSW-only: Number of bits per dimensions iff {@link #HNSW_USE_RABITQ} is set to {@code "true"}, ignored
+     * otherwise. If RaBitQ encoding is used, a vector is stored using roughly
+     * {@code 25 + numDimensions * (numExBits + 1) / 8} bytes. The default value is set to
```

Comment:
Are there really 25 bytes of header per vector, or should that be 25 _bits_?

I also take it that we're following the convention of not counting the sign bit. So 0 should be a legal value here, indicating that we should do binary quantization